### PR TITLE
Moved the reporter functions to a separate file.

### PR DESCRIPTION
**Pull request by come-maiz**
I moved the current reporter to its own file, and added tests to understand and pin down its functionality.
This is first step to be able to add new reporters. The one I'm after is the subunit reporter, but there are other people after xunit and junit, which will also be nice.
Next step is to split CallSuccess and CallSkip.
Thanks to #29.

### CODE REVIEW

File: reporter_test.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
+package check_test
+
+import (
+	"fmt"
+
+	. "gopkg.in/check.v1"
+)
+
+const testFile = "reporter_test.go"
```
Comment (fgimenez) [inline reporter_test.go]: Being a bit pedantic you could get the name of the current file with something like:

```
type ReporterS struct{
  testFile string
}

func (s *ReporterS) SetUpSuite(c *C) {
  _, fileName, _, ok := runtime.Caller(0)
  c.Assert(ok, Equals, true)
  s.testFile = filepath.Base(fileName)
}
```

Moreover, this way the check_test namespace doesn't get a new identifier.

File: reporter_test.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+var _ = Suite(&ReporterS{})
+
+type ReporterS struct{}
```
Comment (fgimenez) [inline reporter_test.go]: Does the suite need to be exported?

**Comment by fgimenez** [comment]
@elopio LGTM, a couple of nits, thanks!

File: reporter_test.go (hunk 2 above)
Comment (come-maiz) [inline reporter_test.go]: it does not, updating.

File: reporter_test.go (hunk 1 above)
Comment (come-maiz) [inline reporter_test.go]: Awesome! that works.

**Comment by come-maiz** [comment]
Thanks @fgimenez. I pushed the changes.

**Comment by come-maiz** [comment]
yay, thanks.

**Pull request merged**